website/blog/2025-07-01-mlflow-prompt-evaluate/index.md

```diff
@@ -0,0 +1,435 @@
+
+Results are all logged in the MLflow run, which can be accessed through the MLflow UI or programatically.
+
+We will see MLflow Evaluate in action as we walk through our OCR use case.
+
+### Additional Concepts
+
+While not central to this example, there are a few other MLflow functionalities that are critical to our GenAI application development lifecycle.
+
+**Tracking:**
+[MLflow Tracking](https://mlflow.org/docs/latest/ml/tracking) allows you to log, organize, and visualize experiment results. This makes our result set more ingestable and auditable.
+
+**Tracing:**
+[MLflow Tracing](https://mlflow.org/docs/latest/genai/tracing) takes tracking one step further in order to have end-to-end observability for GenAI workflows. Again, this keeps everything organized as we work through and iterate on our workflows.
+
+**Autolog**
+Another extension of tracking is [automatic logging](https://mlflow.org/docs/latest/ml/tracking/autolog). This will simplify the tracking process by reducing the need for manual log statements. Metrics, params, artifacts, and other useful information will be logged automatically.
+
+See [Further Reading](#further-reading) for additional blogs digging into these concepts.
+
+## Our Use Case
+
+Our data consists of scanned documents and their corresponding text extracted as JSON. Like mentioned, our goal is to utilize LLMs to build a tool that will handle the text extraction (OCR) for us.
+
+## Setting it Up
+
+We require the following packages
+
+```
+openai
+mlflow
+tiktoken
+aiohttp
+```
+
+which can be installed in the requirement.txt file:
+
+```bash
+pip install -r requirements.txt -qU
+```
+
+<br/>
+Since we will need to make calls to OpenAI's LLMs, we can utilize helper functions to read the OpenAI API key that is stored in a (known) secure location. We can then set the environment variable ```OPENAI_API_KEY```.
+
+```python
+import os
+
+def _set_openai_api_key_for_demo() -> bool:
+    """Dummy method for doing easy auth in a notebook environment during the live demo."""
+    try:
+        with open("/XXXX/openai_api_key.txt", "r") as f: #replace this with your actual directory
+            os.environ["OPENAI_API_KEY"] = f.read().strip()
+            return True
+    except Exception:
+        pass
+
+    return False
+```
+
+Alternatively, we can prompt the user to type the OpenAI API key without echoing using `getpass()`:
+
+```python
+import os
+from utils import _set_openai_api_key_for_demo
+
+if (not _set_openai_api_key_for_demo()) and (not os.getenv("OPENAI_API_KEY")):
+    os.environ["OPENAI_API_KEY"] = getpass("Your OpenAI API Key: ")
+```
+
+### Explore Our Data
+
+We are going to use the [FUNSD dataset](https://guillaumejaume.github.io/FUNSD/) for this exercise, which contains around 200 fully annotated forms, in the form of semantic entity labels and word groupings as shown in the example below:
```

Comment:
Deduplicate redunant info from this comment: https://github.com/mlflow/mlflow-website/pull/321/files#r2167713182